### PR TITLE
[WIP] Change StatusBar color to darker shade of stream color on narrow.

### PR DESCRIPTION
**Pull request by vishwesh3**
Fix:#435

f8076979-e461-479e-92d7-22945c6d45f9
dafd1407-3541-4fb1-a58a-d1050c2a0da9
82780d2e-58b2-41ab-8e13-6b0cdc7244e9

### CODE REVIEW

File: src/utils/color.js [hunk 1]
```diff
@@ -1,2 +1,24 @@
-export const foregroundColorFromBackground = (color: string) =>
+const foregroundColorFromBackground = (color: string) =>
```
Comment (borisyankov) [inline src/utils/color.js]: You can have two exports on each function.

File: src/utils/color.js [hunk 2]
```diff
@@ -1,2 +1,24 @@
...
   (parseInt(color[1], 16) + parseInt(color[3], 16) + parseInt(color[5], 16) > 23 ? 'black' : 'white');
+
+function shadeColor(color, percent) {
```
Comment (borisyankov) [inline src/utils/color.js]: We should just use a library that does that already.
The best I could find is:
https://github.com/Qix-/color

**Comment by borisyankov** [comment]
That looks better :)

Two things:
1. Extract the background logic to some function in our own color.js file. (maybe call it getStatusBar()). This way we make sure the dependencies on external libs are localized.
2. I am not sure if you need to check for platform. I think on iOS the parameter is ignored. Check if true, if yes, don't even do the check.

**Comment by vishwesh3** [comment]
Updated.
Why to first convert color from RGB to HSL and then manipulate and again convert back to RGB.
Found a quick solution by Craig Buckler.
https://www.sitepoint.com/javascript-generate-lighter-darker-color/

**Comment by borisyankov** [comment]
I was going to merge the previous version ;)
Not sure why you decide to replace the code with this snippet.
I am definitely not a fan of code copy-pasted from places on the internet vs supported library.

**Comment by borisyankov** [comment]
Implement against this PR #499
I did add [Qix-/color](https://github.com/Qix-/color)

**Comment by vishwesh3** [comment]
Implemented (in #499)

**Pull request closed without merging**